### PR TITLE
(RE-8835) Restructure artifact paths before shipping

### PR DESCRIPTION
**Pull request by melissa**
Here we have another exercise in restraint. However, this time I didn't remove any functionality, I only moved things around into a more logical (in my opinion) format.

This PR makes a lot of changes in how we're prepping artifacts for shipping, and in how we ship artifacts. I've tested the full invocation of `rake pl:jenkins:uber_ship`, but I did skip signing. 

Please note, this change breaks setting `FOSS_ONLY=true` for those artifact refs that have the old style (note PC1 and earlier era) path format. The automation will be unable to find and download those artifacts for staging.

### CODE REVIEW

**Comment by shrug** [comment]
I think this needs more labels

**Comment by melissa** [comment]
To test this, I checked out a fresh copy of puppet-agent, boot strapped into this working branch of the packaging repo, and ran the following command

```
REPO_NAME='puppet5-nightly' \
DMG_PATH='/opt/downloads' \
MSI_PATH='/opt/downloads' \
SWIX_PATH='/opt/downloads' \
MSI_STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
YUM_HOST='nudc38wcfjukp20.delivery.puppetlabs.net' \
DMG_HOST='nudc38wcfjukp20.delivery.puppetlabs.net' \
TAR_HOST='nudc38wcfjukp20.delivery.puppetlabs.net' \
APT_HOST='nudc38wcfjukp20.delivery.puppetlabs.net' \
APT_SIGNING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
APT_STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
DMG_STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
MSI_HOST='nudc38wcfjukp20.delivery.puppetlabs.net' \
SWIX_HOST='nudc38wcfjukp20.delivery.puppetlabs.net' \
YUM_STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
SWIX_STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
TAR_STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
MSI_STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
STAGING_SERVER='nudc38wcfjukp20.delivery.puppetlabs.net' \
rake pl:jenkins:uber_ship --trace
```

I first tried it with `FOSS_ONLY=true`, but this failed to download all the artifacts. I removed that value, and once the retrieve was finished, I canceled the ship. I then went it and removed the PE only artifacts (aix, solaris, and a few others), and reran the above command.

I skipped the retrieve (as I already had packages staged in `pkg`) and I skipping the signing (as I have not made any required modifications to these tasks yet). I proceeded with all the actual package ships up until the repo update tasks. At which point, I canceled the rake process.

On my test server, this is the contents of `/opt/repository`, `/opt/tools/freight`, and `/opt/downloads` after the main ship jobs had completed.

```
[root@nudc38wcfjukp20 ~]# find /opt/repository /opt/tools /opt/downloads -type f | sort
/opt/downloads/eos/puppet5-nightly/4/i386/puppet-agent-4.99.0.151.gb9d4d03-1.eos4.i386.swix
/opt/downloads/mac/puppet5-nightly/10.10/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.osx10.10.dmg
/opt/downloads/mac/puppet5-nightly/10.11/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.osx10.11.dmg
/opt/downloads/mac/puppet5-nightly/10.12/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.osx10.12.dmg
/opt/downloads/windows/puppet5-nightly/puppet-agent-4.99.0.151.gb9d4d03-x64.msi
/opt/downloads/windows/puppet5-nightly/puppet-agent-4.99.0.151.gb9d4d03-x86.msi
/opt/repository/yum/puppet5-nightly/cisco-wrlinux/5/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.cisco_wrlinux5.x86_64.rpm
/opt/repository/yum/puppet5-nightly/cisco-wrlinux/7/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.cisco_wrlinux7.x86_64.rpm
/opt/repository/yum/puppet5-nightly/el/5/i386/puppet-agent-4.99.0.151.gb9d4d03-1.el5.i386.rpm
/opt/repository/yum/puppet5-nightly/el/5/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.el5.x86_64.rpm
/opt/repository/yum/puppet5-nightly/el/6/i386/puppet-agent-4.99.0.151.gb9d4d03-1.el6.i386.rpm
/opt/repository/yum/puppet5-nightly/el/6/s390x/puppet-agent-4.99.0.151.gb9d4d03-1.el6.s390x.rpm
/opt/repository/yum/puppet5-nightly/el/6/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.el6.x86_64.rpm
/opt/repository/yum/puppet5-nightly/el/7/s390x/puppet-agent-4.99.0.151.gb9d4d03-1.el7.s390x.rpm
/opt/repository/yum/puppet5-nightly/el/7/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.el7.x86_64.rpm
/opt/repository/yum/puppet5-nightly/fedora/f24/i386/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf24.i386.rpm
/opt/repository/yum/puppet5-nightly/fedora/f24/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf24.x86_64.rpm
/opt/repository/yum/puppet5-nightly/fedora/f25/i386/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf25.i386.rpm
/opt/repository/yum/puppet5-nightly/fedora/f25/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf25.x86_64.rpm
/opt/repository/yum/puppet5-nightly/sles/11/i386/puppet-agent-4.99.0.151.gb9d4d03-1.sles11.i386.rpm
/opt/repository/yum/puppet5-nightly/sles/11/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.sles11.x86_64.rpm
/opt/repository/yum/puppet5-nightly/sles/12/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.sles12.x86_64.rpm
/opt/tools/freight/apt/deb/cumulus/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1cumulus_amd64.deb
/opt/tools/freight/apt/deb/huaweios/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1huaweios_powerpc.deb
/opt/tools/freight/apt/deb/jessie/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1jessie_amd64.deb
/opt/tools/freight/apt/deb/jessie/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1jessie_i386.deb
/opt/tools/freight/apt/deb/trusty/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1trusty_amd64.deb
/opt/tools/freight/apt/deb/trusty/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1trusty_i386.deb
/opt/tools/freight/apt/deb/wheezy/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1wheezy_amd64.deb
/opt/tools/freight/apt/deb/wheezy/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1wheezy_i386.deb
/opt/tools/freight/apt/deb/xenial/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1xenial_amd64.deb
/opt/tools/freight/apt/deb/xenial/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1xenial_i386.deb
/opt/tools/freight/apt/deb/yakkety/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1yakkety_amd64.deb
/opt/tools/freight/apt/deb/yakkety/puppet5-nightly/puppet-agent_4.99.0.151.gb9d4d03-1yakkety_i386.deb
```

This is the contents of `pkg` after shipping. Note this automation does not change how the packages are staged in `pkg` at all. It does those modifications in a temporary directory
```
$ find pkg -type f | sort
pkg/apple/10.10/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.osx10.10.dmg
pkg/apple/10.11/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.osx10.11.dmg
pkg/apple/10.12/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.osx10.12.dmg
pkg/b9d4d03d9dc2aad54fb855abafe73e7e8003f131.yaml
pkg/cisco-wrlinux/5/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.cisco_wrlinux5.x86_64.rpm
pkg/cisco-wrlinux/7/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.cisco_wrlinux7.x86_64.rpm
pkg/deb/cumulus/PC1/puppet-agent_4.99.0.151.gb9d4d03-1cumulus_amd64.deb
pkg/deb/huaweios/PC1/puppet-agent_4.99.0.151.gb9d4d03-1huaweios_powerpc.deb
pkg/deb/jessie/PC1/puppet-agent_4.99.0.151.gb9d4d03-1jessie_amd64.deb
pkg/deb/jessie/PC1/puppet-agent_4.99.0.151.gb9d4d03-1jessie_i386.deb
pkg/deb/trusty/PC1/puppet-agent_4.99.0.151.gb9d4d03-1trusty_amd64.deb
pkg/deb/trusty/PC1/puppet-agent_4.99.0.151.gb9d4d03-1trusty_i386.deb
pkg/deb/wheezy/PC1/puppet-agent_4.99.0.151.gb9d4d03-1wheezy_amd64.deb
pkg/deb/wheezy/PC1/puppet-agent_4.99.0.151.gb9d4d03-1wheezy_i386.deb
pkg/deb/xenial/PC1/puppet-agent_4.99.0.151.gb9d4d03-1xenial_amd64.deb
pkg/deb/xenial/PC1/puppet-agent_4.99.0.151.gb9d4d03-1xenial_i386.deb
pkg/deb/yakkety/PC1/puppet-agent_4.99.0.151.gb9d4d03-1yakkety_amd64.deb
pkg/deb/yakkety/PC1/puppet-agent_4.99.0.151.gb9d4d03-1yakkety_i386.deb
pkg/el/5/PC1/i386/puppet-agent-4.99.0.151.gb9d4d03-1.el5.i386.rpm
pkg/el/5/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.el5.x86_64.rpm
pkg/el/6/PC1/i386/puppet-agent-4.99.0.151.gb9d4d03-1.el6.i386.rpm
pkg/el/6/PC1/s390x/puppet-agent-4.99.0.151.gb9d4d03-1.el6.s390x.rpm
pkg/el/6/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.el6.x86_64.rpm
pkg/el/7/PC1/s390x/puppet-agent-4.99.0.151.gb9d4d03-1.el7.s390x.rpm
pkg/el/7/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.el7.x86_64.rpm
pkg/eos/4/PC1/i386/puppet-agent-4.99.0.151.gb9d4d03-1.eos4.i386.swix
pkg/fedora/f24/PC1/i386/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf24.i386.rpm
pkg/fedora/f24/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf24.x86_64.rpm
pkg/fedora/f25/PC1/i386/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf25.i386.rpm
pkg/fedora/f25/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.fedoraf25.x86_64.rpm
pkg/puppet-agent-4.99.0-151-signing_bundle.tar.gz
pkg/sles/11/PC1/i386/puppet-agent-4.99.0.151.gb9d4d03-1.sles11.i386.rpm
pkg/sles/11/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.sles11.x86_64.rpm
pkg/sles/12/PC1/x86_64/puppet-agent-4.99.0.151.gb9d4d03-1.sles12.x86_64.rpm
pkg/windows/puppet-agent-4.99.0.151.gb9d4d03-x64.msi
pkg/windows/puppet-agent-4.99.0.151.gb9d4d03-x64.wixpdb
pkg/windows/puppet-agent-4.99.0.151.gb9d4d03-x86.msi
pkg/windows/puppet-agent-4.99.0.151.gb9d4d03-x86.wixpdb
pkg/windows/puppet-agent-x64.msi
pkg/windows/puppet-agent-x86.msi
```

**Comment by melissa** [comment]
@khussey this and https://github.com/puppetlabs/packaging/pull/651 is why automating the puppet5-nightly repos is taking so long :( we're making headway though!!

**Comment by melissa** [comment]
So, Travis is failing on Ruby 1.9.3 and 2.0.0, but passes on Ruby 2.1.1 and on my laptop with Ruby 2.3.3.

I'm really tempted to just update Travis here. Thoughts?

File: lib/packaging/util/ship.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    new_pkgs
+  end
+
+  # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+  def ship_pkgs(pkg_exts, staging_server, remote_path, options = { excludes: [], chattr: true })
+    # First find the packages to be shipped. We must find them before moving
+    # to our temporary staging directory
+    local_packages = collect_packages(pkg_exts, options[:excludes])
+
+    tmpdir = "/tmp/ship-#{Pkg::Util.rand_string}"
```
Comment (underscorgan) [inline lib/packaging/util/ship.rb]: We should use `Dir.mktmpdir` here.

File: lib/packaging/platforms.rb [hunk 2]
```diff
@@ -1,133 +1,224 @@
...
+  # Helper meta-method to return all platforms that use .rpm packages
+  # @scope class
+  formats.each do |format|
+    type = "by_#{format}".to_sym
+    define_method(type) do
+      by_package_format format
     end
   end
+
+  private :by_package_format
```
Comment (underscorgan) [inline lib/packaging/platforms.rb]: This file has a lot of nested looping. Can we simplify this?

**Comment by underscorgan** [comment]
wrt travis failures, it seems reasonable to require ruby >= 2.1 for a 1.0 release, but what are we going to break by doing that? do we still run older rubies in places?

File: lib/packaging/platforms.rb [hunk 3]
```diff
@@ -1,133 +1,224 @@
...
-          r[:package_format]
-        end
-      end
-      fmts.to_set.sort
+    fmts.to_set.sort
+  end
+
+  # @return [Array] An array of Strings, containing all of the supported
+  #   platforms as defined in PLATFORM_INFO
+  def supported_platforms
```
Comment (mwaggett) [inline lib/packaging/platforms.rb]: Is this necessary? Can't you just use `PLATFORM_INFO.keys`?

File: lib/packaging/platforms.rb (hunk 2 above)
Comment (mwaggett) [inline lib/packaging/platforms.rb]: @underscorgan To be clear, you're referring to everything in this file, not a specific function, right?

Comment (underscorgan) [inline lib/packaging/platforms.rb]: yes, it's a generic comment. Probably some of them are okay it just feels like a lot of nested looping so i'd like to see if we can simplify some of that

Comment (mwaggett) [inline lib/packaging/platforms.rb]: 👍 on it

Comment (mwaggett) [inline lib/packaging/platforms.rb]: @underscorgan so nothing is jumping out at me as immediately simplifiable and I'm hesitant to spend a ton of time trying to rework these things, but if you have specific ideas/suggestions, I'd love to hear them!

Comment (underscorgan) [inline lib/packaging/platforms.rb]: I added a few notes, not finding any clear simplifications otherwise.

File: lib/packaging/util/ship.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+require 'tmpdir'
+module Pkg::Util::Ship
+  module_function
+
+  def collect_packages(pkg_exts, excludes = []) # rubocop:disable Metrics/MethodLength
+    pkgs = []
+    pkg_exts.each do |ext|
+      pkgs += Dir.glob(ext)
+    end
+    if excludes
```
Comment (underscorgan) [inline lib/packaging/util/ship.rb]: you don't need this, looping over an empty array doesn't throw errors and will just continue.

File: lib/packaging/util/ship.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
+# Module for shipping all packages to places
+require 'tmpdir'
+module Pkg::Util::Ship
+  module_function
+
+  def collect_packages(pkg_exts, excludes = []) # rubocop:disable Metrics/MethodLength
+    pkgs = []
+    pkg_exts.each do |ext|
+      pkgs += Dir.glob(ext)
+    end
```
Comment (underscorgan) [inline lib/packaging/util/ship.rb]: I think this can be simplified into pkgs = pkg_exts.map { |ext| Dir.glob(ext) }.flatten

**Comment by mwaggett** [comment]
okay, I think this is ready for another round of review. 
I tested the same way Melissa did and got the same result, only without 'deb/' under the apt directory. 
This still fails on early versions of Ruby - what do we want to do about that?

**Comment by underscorgan** [comment]
Tested by running
```
morgan@river.corp.puppetlabs.net % REPO_NAME='puppet5-nightly' \                                                            (git)-[e36cade...]
MSI_STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
YUM_HOST='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
DMG_HOST='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
TAR_HOST='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
APT_HOST='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
APT_SIGNING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
APT_STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
DMG_STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
MSI_HOST='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
SWIX_HOST='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
YUM_STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
SWIX_STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
TAR_STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
MSI_STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
STAGING_SERVER='root@cbopnytpvluze7u.delivery.puppetlabs.net' \
rake pl:jenkins:uber_ship --trace
```

Files in `/opt/` afterwards on my vm:
```
/opt/tools/freight/apt/cumulus/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1cumulus_amd64.deb
/opt/tools/freight/apt/huaweios/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1huaweios_powerpc.deb
/opt/tools/freight/apt/jessie/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1jessie_amd64.deb
/opt/tools/freight/apt/jessie/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1jessie_i386.deb
/opt/tools/freight/apt/trusty/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1trusty_amd64.deb
/opt/tools/freight/apt/trusty/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1trusty_i386.deb
/opt/tools/freight/apt/wheezy/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1wheezy_amd64.deb
/opt/tools/freight/apt/wheezy/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1wheezy_i386.deb
/opt/tools/freight/apt/xenial/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1xenial_amd64.deb
/opt/tools/freight/apt/xenial/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1xenial_i386.deb
/opt/tools/freight/apt/yakkety/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1yakkety_amd64.deb
/opt/tools/freight/apt/yakkety/puppet5-nightly/puppet-agent_4.99.0.202.ge36cade-1yakkety_i386.deb
/opt/repository/yum/puppet5-nightly/cisco-wrlinux/5/x86_64/puppet-agent-4.99.0.202.ge36cade-1.cisco_wrlinux5.x86_64.rpm
/opt/repository/yum/puppet5-nightly/cisco-wrlinux/7/x86_64/puppet-agent-4.99.0.202.ge36cade-1.cisco_wrlinux7.x86_64.rpm
/opt/repository/yum/puppet5-nightly/el/5/i386/puppet-agent-4.99.0.202.ge36cade-1.el5.i386.rpm
/opt/repository/yum/puppet5-nightly/el/5/x86_64/puppet-agent-4.99.0.202.ge36cade-1.el5.x86_64.rpm
/opt/repository/yum/puppet5-nightly/el/6/i386/puppet-agent-4.99.0.202.ge36cade-1.el6.i386.rpm
/opt/repository/yum/puppet5-nightly/el/6/x86_64/puppet-agent-4.99.0.202.ge36cade-1.el6.x86_64.rpm
/opt/repository/yum/puppet5-nightly/el/7/x86_64/puppet-agent-4.99.0.202.ge36cade-1.el7.x86_64.rpm
/opt/repository/yum/puppet5-nightly/fedora/f24/i386/puppet-agent-4.99.0.202.ge36cade-1.fedoraf24.i386.rpm
/opt/repository/yum/puppet5-nightly/fedora/f24/x86_64/puppet-agent-4.99.0.202.ge36cade-1.fedoraf24.x86_64.rpm
/opt/repository/yum/puppet5-nightly/fedora/f25/i386/puppet-agent-4.99.0.202.ge36cade-1.fedoraf25.i386.rpm
/opt/repository/yum/puppet5-nightly/fedora/f25/x86_64/puppet-agent-4.99.0.202.ge36cade-1.fedoraf25.x86_64.rpm
/opt/repository/yum/puppet5-nightly/sles/11/i386/puppet-agent-4.99.0.202.ge36cade-1.sles11.i386.rpm
/opt/repository/yum/puppet5-nightly/sles/11/x86_64/puppet-agent-4.99.0.202.ge36cade-1.sles11.x86_64.rpm
/opt/repository/yum/puppet5-nightly/sles/12/x86_64/puppet-agent-4.99.0.202.ge36cade-1.sles12.x86_64.rpm
/opt/downloads/mac/puppet5-nightly/10.10/x86_64/puppet-agent-4.99.0.202.ge36cade-1.osx10.10.dmg
/opt/downloads/mac/puppet5-nightly/10.11/x86_64/puppet-agent-4.99.0.202.ge36cade-1.osx10.11.dmg
/opt/downloads/mac/puppet5-nightly/10.12/x86_64/puppet-agent-4.99.0.202.ge36cade-1.osx10.12.dmg
/opt/downloads/eos/puppet5-nightly/4/i386/puppet-agent-4.99.0.202.ge36cade-1.eos4.i386.swix
/opt/downloads/windows/puppet5-nightly/puppet-agent-4.99.0.202.ge36cade-x64.msi
/opt/downloads/windows/puppet5-nightly/puppet-agent-4.99.0.202.ge36cade-x86.msi
```

Files in `pkg` locally:
```
pkg/apple/10.10/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.osx10.10.dmg
pkg/apple/10.11/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.osx10.11.dmg
pkg/apple/10.12/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.osx10.12.dmg
pkg/cisco-wrlinux/5/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.cisco_wrlinux5.x86_64.rpm
pkg/cisco-wrlinux/7/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.cisco_wrlinux7.x86_64.rpm
pkg/deb/cumulus/PC1/puppet-agent_4.99.0.202.ge36cade-1cumulus_amd64.deb
pkg/deb/huaweios/PC1/puppet-agent_4.99.0.202.ge36cade-1huaweios_powerpc.deb
pkg/deb/jessie/PC1/puppet-agent_4.99.0.202.ge36cade-1jessie_amd64.deb
pkg/deb/jessie/PC1/puppet-agent_4.99.0.202.ge36cade-1jessie_i386.deb
pkg/deb/trusty/PC1/puppet-agent_4.99.0.202.ge36cade-1trusty_amd64.deb
pkg/deb/trusty/PC1/puppet-agent_4.99.0.202.ge36cade-1trusty_i386.deb
pkg/deb/wheezy/PC1/puppet-agent_4.99.0.202.ge36cade-1wheezy_amd64.deb
pkg/deb/wheezy/PC1/puppet-agent_4.99.0.202.ge36cade-1wheezy_i386.deb
pkg/deb/xenial/PC1/puppet-agent_4.99.0.202.ge36cade-1xenial_amd64.deb
pkg/deb/xenial/PC1/puppet-agent_4.99.0.202.ge36cade-1xenial_i386.deb
pkg/deb/yakkety/PC1/puppet-agent_4.99.0.202.ge36cade-1yakkety_amd64.deb
pkg/deb/yakkety/PC1/puppet-agent_4.99.0.202.ge36cade-1yakkety_i386.deb
pkg/e36cade45d32f0cca85dd3e51b1fcc17e7cd8bcc.yaml
pkg/el/5/PC1/i386/puppet-agent-4.99.0.202.ge36cade-1.el5.i386.rpm
pkg/el/5/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.el5.x86_64.rpm
pkg/el/6/PC1/i386/puppet-agent-4.99.0.202.ge36cade-1.el6.i386.rpm
pkg/el/6/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.el6.x86_64.rpm
pkg/el/7/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.el7.x86_64.rpm
pkg/eos/4/PC1/i386/puppet-agent-4.99.0.202.ge36cade-1.eos4.i386.swix
pkg/fedora/f24/PC1/i386/puppet-agent-4.99.0.202.ge36cade-1.fedoraf24.i386.rpm
pkg/fedora/f24/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.fedoraf24.x86_64.rpm
pkg/fedora/f25/PC1/i386/puppet-agent-4.99.0.202.ge36cade-1.fedoraf25.i386.rpm
pkg/fedora/f25/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.fedoraf25.x86_64.rpm
pkg/puppet-agent-4.99.0-202-signing_bundle.tar.gz
pkg/sles/11/PC1/i386/puppet-agent-4.99.0.202.ge36cade-1.sles11.i386.rpm
pkg/sles/11/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.sles11.x86_64.rpm
pkg/sles/12/PC1/x86_64/puppet-agent-4.99.0.202.ge36cade-1.sles12.x86_64.rpm
pkg/windows/puppet-agent-4.99.0.202.ge36cade-x64.msi
pkg/windows/puppet-agent-4.99.0.202.ge36cade-x64.wixpdb
pkg/windows/puppet-agent-4.99.0.202.ge36cade-x86.msi
pkg/windows/puppet-agent-4.99.0.202.ge36cade-x86.wixpdb
pkg/windows/puppet-agent-x64.msi
pkg/windows/puppet-agent-x86.msi
```

**Comment by shrug** [comment]
We should remove the obsoleted rubies from the travis config

File: lib/packaging/paths.rb [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+#
+# rubocop:disable Metrics/ModuleLength
+module Pkg::Paths
+  include Pkg::Platforms
+
+  module_function
+
+  # Given a path to an artifact, divine the appropriate platform tag associated
+  # with the artifact and path
+  def tag_from_artifact_path(path)
```
Comment (underscorgan) [inline lib/packaging/paths.rb]: This doesn't work for all tags, looks like it's somewhat different from the version I had working.

It works for foss-only platforms but this method should work for all currently supported platforms. I'll get changes pushed up.

Comment (underscorgan) [inline lib/packaging/paths.rb]: OK, fixed in 2512ddd260eda2749ef3020fbb78f612831c6116

**Comment by underscorgan** [comment]
@mwaggett want to rebase this against 1.0.x to pick up the travis changes?

**Comment by mwaggett** [comment]
@underscorgan rebased

**Pull request merged**